### PR TITLE
feat(sample-data): add sample database generator for testing and local dev setup

### PR DESCRIPTION
**Pull request by nsmela**
Added a small-site sample database generator for developers and debugging.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (adds functionality, always assumed to be breaking until tested by maintainers)
- [ ] Breaking change (fix or addition that changes existing behaviour)
- [ ] Documentation only

## Target Branch

`develop`

## AI Usage Disclosure

- [ ] No AI tools were used in this PR.
- [x] AI tools were used:
  - **Tools / where used:** Claude / Gemini for authoring sample data generation script and test fixtures.
  - [x] I've reviewed every AI-generated line as if I wrote it myself, and confirm: no PHI shared, no license-incompatible content (Apache-2.0).

---

### Coder Tasks

- [x] Rebased / merged `develop` branch; no merge conflicts.
- [x] `uv run pre-commit run --all-files` passes (ruff lint, ruff-format, django-upgrade).
  - `uv run ruff check . --fix` to help resolve linting errors
- [x] `uv run python -Wd manage.py check` reviewed for new deprecation warnings.
- [x] `uv run pytest` full test suite passes.
- [x] `uv run python manage.py makemigrations --check --dry-run` if migrations are found, change should be marked Breaking
- [x] Docker builds succeed if touched.
- [x] Docs updated under `docs/` if behaviour, settings, or APIs changed.
- [x] `release_notes.md` updated if user-facing.
- [x] No secrets, debug code, or stray `print`/`console.log` left in.

### Review Code

- [ ] Reviewer verifies the target branch.
- [ ] For non-trivial changes: pulled the branch, ran it locally, and confirmed it works as described.
- [ ] CI is green; test suite and `pre-commit` pass.
- [ ] Reads through all changed files for anything weird or unintended.
- [ ] Logic is correct; edge cases and error handling considered.
- [ ] Migrations are sane and reversible where possible.
- [ ] Changes are consistent with the modernization direction (no reintroduction
      of patterns being migrated away from, e.g. flake8/yapf-era conventions).
- [ ] No security concerns (input validation, permissions, injection, XSS).
- [ ] Comments are sufficient and not overkill.
- [ ] Typos, spellcheck, professional.
- [ ] Reviewer sends back review.

### Edits

- [ ] Coder fixes any issues.
- [ ] Reviewer approves.

---

## Testing Notes

- Tested via `uv` on local Windows environment with Python 3.12.
- Verified command execution:
  ```bash
  uv run python manage.py generate_sample_data --days=90 --clear --no-input
  ```
- Ran automated test suite:
  ```bash
  uv run pytest qatrack/qatrack_core/tests/test_sample_data.py
  ```
  Result: `10 passed in 22.51s`

  Covers the generated dataset, ad-hoc (unscheduled) return to service test
  lists, rejection of negative `--days`, re-running without `--clear` (no
  duplicated data, no unit number collision, existing configuration kept),
  existing and externally linked user accounts, partially loaded default
  fixtures, and rollback when generation fails after `--clear`.
- `uv run pre-commit run --all-files` passes; the docs build adds no new warnings.
- Manually tested UI workflows on Chrome:
  - QA Perform & In-Progress resumption on `Versa-HD`.
  - Review queue signoff as `jane.physicist`.
  - Service Log event details and RTS QA viewing.
  - Parts inventory low-stock alerts.
  - PDF and Excel report generation under `/reports/`.

## Screenshots

N/A (Backend management command & test data generator)

```

### CODE REVIEW

**Comment by nsmela** [comment]
Was going to wait to submit this, but it's been super-useful when validating the recent PRs, so I figured it was best to share and hopefully make your life easier. Will require ammending docs for developers, contributors, etc

File: qatrack/qatrack_core/sample_data/small.py [hunk 1]
```diff
@@ -0,0 +1,1265 @@
...
+                    'is_staff': is_staff,
+                    'is_superuser': is_superuser,
+                }
+            )
+            user.set_password("password123")
+            user.save()
+            user.groups.set(groups)
+            return user
+
+        self.user_admin = create_user("admin", "admin@qatrackplus.com", "System", "Admin", [self.grp_admins], is_superuser=True)
```
[the remark this answers is not in the document]
Comment (crcrewso) [reply qatrack/qatrack_core/sample_data/small.py:146]: @coderabbitai pretty sure this is intended behavior, what happens if the new linked account does not have a password?

File: qatrack/qatrack_core/management/commands/generate_sample_data.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        parser.add_argument(
+            "--size",
+            type=str,
+            default="small",
+            choices=["small", "medium", "large"],
+            help="Size and profile of the radiation oncology center to generate (default: small).",
+        )
+        parser.add_argument(
+            "--days",
+            type=int,
```
[the remark this answers is not in the document]
Comment (crcrewso) [reply qatrack/qatrack_core/management/commands/generate_sample_data.py:20]: negative days should not be allowed on principle,

File: qatrack/qatrack_core/management/commands/generate_sample_data.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+class Command(BaseCommand):
+    help = "Generates comprehensive, realistic sample test data for QATrack+ (QA, Service Log, Faults, Parts, Reports)."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--size",
+            type=str,
+            default="small",
+            choices=["small", "medium", "large"],
```
[the remark this answers is not in the document]
Comment (crcrewso) [reply qatrack/qatrack_core/management/commands/generate_sample_data.py:15]: I'm okay with leaving the options here as long as we document only small is implemented. 

alternatively trap the CommandError and raise a specific NotImplemented error, 

dev's choice.

**Review by crcrewso — CHANGES_REQUESTED** [review]

File: qatrack/qatrack_core/management/commands/generate_sample_data.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nsmela) [reply qatrack/qatrack_core/management/commands/generate_sample_data.py:15]: I'll try implementing a medium and large dataset for this PR. Wanted to verify this approach is okay before working further on it.

[the remark this answers is not in the document]
Comment (crcrewso) [reply qatrack/qatrack_core/management/commands/generate_sample_data.py:15]: I don't think that's necessary. If you want to sure, but I think this is so handy that sooner is better than complete.

[the remark this answers is not in the document]
Comment (nsmela) [reply qatrack/qatrack_core/management/commands/generate_sample_data.py:15]: Kk. I'll focus on removing the option for medium and large sets then. It'll be trivial to change the command to implement them later.

[the remark this answers is not in the document]
Comment (nsmela) [reply qatrack/qatrack_core/management/commands/generate_sample_data.py:15]: @coderabbitai remove the sizing options (small, medium, large) and have this small dataset be the only option

[the remark this answers is not in the document]
Comment (nsmela) [reply qatrack/qatrack_core/management/commands/generate_sample_data.py]: Option has been removed for now. A small Centre is the default until more are added

**Comment by nsmela** [comment]
Ready for review

File: qatrack/qatrack_core/sample_data/small.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nsmela) [reply qatrack/qatrack_core/sample_data/small.py:164]: accounts the generator creates still get password123, and an existing account gets it too when has_usable_password() is False
